File: recursive_search.py
import os

def search_recursive(folder_path, target):
    # Duyệt qua từng thứ trong thư mục
    for name in os.listdir(folder_path):
        full_path = os.path.join(folder_path, name)

        if os.path.isfile(full_path) and name == target:
            print(f"Tim thay file: {full_path}")
            return True

        if os.path.isdir(full_path):
            # Gọi đệ quy vào thư mục con
            if search_recursive(full_path, target):
                return True

    return False

File: stack_c++/recursive_search.cpp
#include "recursive_search.h"
#include <filesystem>
#include <iostream>

namespace fs = std::filesystem;

// Tuong duong voi search_recursive() trong recursive_search.py
bool searchRecursive(const std::string &folder_path, const std::string &target)
{
    // Duyet qua tung thu trong thu muc
    for (const auto &entry : fs::directory_iterator(folder_path))
    {
        const fs::path &full_path = entry.path();
        std::string name = full_path.filename().string();

        if (entry.is_regular_file() && name == target)
        {
            std::cout << "Tim thay file: " << full_path.string() << std::endl;
            return true;
        }

        if (entry.is_directory())
        {
            // Goi de quy vao thu muc con
            if (searchRecursive(full_path.string(), target))
            {
                return true;
            }
        }
    }
    return false;
}